src/plugins/intel_cpu/src/emitters/tpp/x64/jit_eltwise_emitters.hpp

```diff
@@ -0,0 +1,120 @@
+
+protected:
+    libxsmm_meltw_binary_shape m_shape;
+    libxsmm_meltw_binary_type m_op_type;
+    void validate_arguments(const std::vector<size_t> &in, const std::vector<size_t> &out) const override;
+    static  libxsmm_blasint get_broadcasted_dim(libxsmm_blasint dim0, libxsmm_blasint dim1, std::pair<bool, bool>& bcast_flags);
+};
+//#define DEBUG_TPP_EMITTERS
+class UnaryEltwiseTppEmitter : public TppEmitter {
+public:
+    UnaryEltwiseTppEmitter(dnnl::impl::cpu::x64::jit_generator* h,
+                            dnnl::impl::cpu::x64::cpu_isa_t isa,
+                            const ov::snippets::lowered::ExpressionPtr& expr);
+    size_t get_inputs_num() const override { return 1; }
+
+    static void execute_kernel(libxsmm_meltwfunction_unary eltwise_kernel, void *in0, void *out0);
+    const uintptr_t get_compiled_kernel_ptr() const override {
+        return reinterpret_cast<const uintptr_t>(libxsmm_dispatch_meltw_unary_v2(m_op_type,
+                                                                                 m_shape,
+                                                                                 m_compile_flags));
+    }
+    const uintptr_t get_execute_function_ptr() const override { return reinterpret_cast<const uintptr_t>(execute_kernel); }
+    static std::set<std::vector<element::Type>> get_supported_precisions(const std::shared_ptr<ov::Node>& node = nullptr);
+
+protected:
+    libxsmm_meltw_unary_shape m_shape;
+    libxsmm_meltw_unary_type m_op_type;
+    void validate_arguments(const std::vector<size_t> &in, const std::vector<size_t> &out) const override;
+};
+
+class ReduceTppEmitter : public UnaryEltwiseTppEmitter {
+public:
+    ReduceTppEmitter(dnnl::impl::cpu::x64::jit_generator* h,
+                     dnnl::impl::cpu::x64::cpu_isa_t isa,
+                     const ov::snippets::lowered::ExpressionPtr& expr);
+};
+
+class ReferenceUnaryEltwiseTppEmitter : public UnaryEltwiseTppEmitter {
+public:
+    // Note: can create template to suppport different executor signatures
+    typedef std::function<float(float)> executor_function;
+    ReferenceUnaryEltwiseTppEmitter(dnnl::impl::cpu::x64::jit_generator* h,
+                                    dnnl::impl::cpu::x64::cpu_isa_t isa,
+                                    const ov::snippets::lowered::ExpressionPtr& expr,
+                                    executor_function executor) :
+                                    UnaryEltwiseTppEmitter(h, isa, expr), executor(std::move(executor)) {
+    }
+    static void execute_unary_eltw_kernel(ReferenceUnaryEltwiseTppEmitter* ref_emitter, void *in0, void *out0) {
+        assert(ref_emitter);
+        // Note: we can instantiate template with different precision combinations here, if we need to
+        // Note: if switch(precision) is too expensive here, we can do it in the constructor and save the selected impl in function ptr
```

Comment:
Ok, thanks for your comment. I just wanted to highlight alternative options for discussion.
Anyway, since we don't have the `switch` (only fp32 is supported), I removed the comment to avoid confusion.

But we can easily support other precisions in future.